nimble/host/include/host/ble_cs.h

```diff
@@ -0,0 +1,352 @@
+    uint8_t snr_control_initiator;
+    uint8_t snr_control_reflector;
+} __attribute__((packed));
+struct ble_cs_set_proc_params_rp {
+    uint8_t status;
+    uint16_t conn_handle;
+} __attribute__((packed));
+
+struct ble_cs_proc_enable_cp {
+    uint16_t conn_handle;
+    uint8_t config_id;
+    uint8_t enable;
+} __attribute__((packed));
+struct ble_cs_proc_enable_complete {
+    uint8_t status;
+    uint16_t conn_handle;
+    uint8_t config_id;
+    uint8_t state;
+    uint8_t tone_antenna_config_selection;
+    uint8_t selected_tx_power;
+    uint8_t subevent_len[3];
+    uint8_t subevents_per_event;
+    uint16_t subevent_interval;
+    uint16_t event_interval;
+    uint16_t procedure_interval;
+    uint16_t procedure_count;
+} __attribute__((packed));
+struct ble_cs_subevent_result {
+    uint16_t conn_handle;
+    uint8_t config_id;
+    uint16_t start_acl_conn_event_counter;
+    uint16_t procedure_counter;
+    uint16_t frequency_compensation;
+    uint8_t reference_power_level;
+    uint8_t procedure_done_status;
+    uint8_t subevent_done_status;
+    uint8_t abort_reason;
+    uint8_t num_antenna_paths;
+    uint8_t num_steps_reported;
+    uint8_t steps[];
+} __attribute__((packed));
+struct ble_cs_subevent_result_continue {
+    uint16_t conn_handle;
+    uint8_t config_id;
+    uint8_t procedure_done_status;
+    uint8_t subevent_done_status;
+    uint8_t abort_reason;
+    uint8_t num_antenna_paths;
+    uint8_t num_steps_reported;
+    uint8_t steps[];
+} __attribute__((packed));
+
+struct ble_cs_test_cp {
+    uint8_t main_mode_type;
+    uint8_t sub_mode_type;
+    uint8_t main_mode_repetition;
+    uint8_t mode_0_steps;
+    uint8_t role;
+    uint8_t rtt_type;
+    uint8_t cs_sync_phy;
+    uint8_t cs_sync_antenna_selection;
+    uint8_t subevent_len[3];
+    uint16_t subevent_interval;
+    uint8_t transmit_power_level;
+    uint8_t t_ip1_time;
+    uint8_t t_ip2_time;
+    uint8_t t_fcs_time;
+    uint8_t t_pm_time;
+    uint8_t t_sw_time;
+    uint8_t tone_antenna_config_selection;
+    uint8_t companion_signal_enable;
+    uint16_t drbg_nonce;
+    uint16_t override_config;
+    uint8_t override_parameters_length;
+    uint8_t override_parameters_data[];
+} __attribute__((packed));
+struct ble_cs_test_rp {
+    uint8_t status;
+} __attribute__((packed));
+
+/**
+ * CS event callbacks.
+ */
+struct ble_cs_callbacks {
+    void (*rd_rem_supp_cap_cb)(struct ble_cs_rem_supp_cap_complete *ev);
+    void (*rd_rem_fae_cb)(struct ble_cs_rd_rem_fae_complete *ev);
+    void (*sec_enable_cb)(struct ble_cs_sec_enable_complete *ev);
+    void (*config_cb)(struct ble_cs_config_complete *ev);
+    void (*proc_enable_cb)(struct ble_cs_proc_enable_complete *evt);
+    void (*subevent_result_cb)(struct ble_cs_subevent_result *evt);
+    void (*subevent_result_continue_cb)(struct ble_cs_subevent_result_continue *evt);
+    void (*test_end_cb)(int status);
+};
+
+void ble_cs_register_callbacks(struct ble_cs_callbacks *cbs);
```

Comment:
for consistency we should better use similar handling of events as we have for e.g. GAP, i.e. add handler to the function that starts procedure and use event struct to pass different events